homeassistant/components/hive/sensor.py

```diff
@@ -1,20 +1,54 @@
 """Support for the Hive sensors."""
 from datetime import timedelta
 
-from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, POWER_KILO_WATT
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HiveEntity
 from .const import DOMAIN
 
 PARALLEL_UPDATES = 0
 SCAN_INTERVAL = timedelta(seconds=15)
-DEVICETYPE = {
-    "Battery": {"unit": " % ", "type": SensorDeviceClass.BATTERY},
-}
+
+
+SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(key="Heating_Mode", icon="mdi:radiator"),
+    SensorEntityDescription(key="Hotwater_Mode", icon="mdi:water-pump"),
+    SensorEntityDescription(key="Hotwater_Boost", icon="mdi:water-pump"),
+    SensorEntityDescription(key="Mode", icon="mdi:eye"),
+    SensorEntityDescription(
+        key="Battery",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Availability",
+        icon="mdi:check-circle",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Connectivity",
+        icon="mdi:check-circle",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="Power",
+        native_unit_of_measurement=POWER_KILO_WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ENERGY,
```

Comment:
If this measures kW, it's not an energy sensor but a power sensor.

Reply:
@MartinHjelmare i updated the sensor to a power type.

Reply:
Did you forget to push the commit? I don't see any changes.

Reply:
@MartinHjelmare  ye I have had some issues with this branch so I have create a new branch and PR #72309 to continue this ill close this PR and tag you in the other one to continue the review.